app/api/v1/routes/blockchain.py

```diff
@@ -95,28 +95,40 @@ async def getInfo():
         logging.error(f'ERR:{myself()}: invalid blockchain info ({e})')
         return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content=f'ERR:{myself()}: invalid blockchain info ({e})')
 
-
 @r.get("/tokenomics/{tokenId}", name="blockchain:tokenomics")
 async def tokenomics(tokenId):
-    engDanaides = create_engine(CFG.csDanaides)
-    sql = text(f'''
-        select t.name as token_name
-            , t.token_price * t.in_circulation as market_cap
-            , emission_amount/power(10, decimals) as initial_total_supply
-            , t.current_total_supply
-            , emission_amount/power(10, decimals) - t.current_total_supply as burned
-            , t.in_circulation
-            , t.token_price
-        from tokens t on t.token_id = a.token_id
-        where t.token_id = :token_id
-    ''')
-    stats = engDanaides.execute(sql, {'token_id': tokenId}).fetchall()
-
-    res = {}
-    for stat in stats:        
-        res[stat['notes']] = stat['amount']
-
-    return res
+    try:
+        engDanaides = create_engine(CFG.csDanaides)
```

Comment:
Suggestion: Have this cached even if the endpoint only takes a couple of seconds to reduce load on db and server
```python
cached = cache.get(f"get_api_blockchain_tokenomics_{tokenId}")
        if cached:
            return cached
```

and 

```python
cache.set(f"get_api_blockchain_tokenomics_{tokenId}", stats)
```

Reply:
https://github.com/ergo-pad/ergopad-api/pull/348